### ISSUE TITLE
CSoundEngine::PrecacheSound does not support sound replacement

### ISSUE THREAD

**Issue by fnky** [baso88/SC_AngelScript#28]
[`CSoundEngine::PrecacheSound`](https://baso88.github.io/SC_AngelScript/docs/CSoundEngine.htm) does not support sound replacement. Sound replacement works by checking the current entity's sound replacement file, which it cannot do without the entity instance. Add an overload that takes a [`CBaseEntity@`](https://baso88.github.io/SC_AngelScript/docs/CBaseEntity.htm) so custom entities can pass in "`self`".

**Comment by fnky** [comment]
Code to add this:

```cpp
void CASSoundEngine::PrecacheSound( CBaseEntity* pEntity, const CString& szFileName )
{
    if ( szFileName.IsEmpty() )
        return;

    PrecacheSound_Intercept( pEntity, const_cast<char*>( STRING( ALLOC_STRING( szFileName.CStr() ) ) ) );
}
```

Declaration and registration in the API should be obvious.

**Comment by baso88** [comment]
The fix for this issue will be included in the next update (5.13). Closing.
